### ISSUE TITLE
coredump occurred in snapshot_operation nemesis

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#5789]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20200209.bcc46475522
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0b464ba1fef4f6228 (aws region: eu-west-1)
Detected by: longevity-3h-10gb
Test-id: d5361982-d2a0-43cc-bc40-b0e725a7245e

### Description:

Coredump occurred on longevity-10gb-3h-master-db-node-d5361982-3 during snapshot_operation Nemesis, the snapshot operation was executed on this node, it's the target node.

Nemesis code: https://github.com/scylladb/scylla-cluster-tests/blob/165b49c1f32b14136592e430a1933e5dbd3e62a2/sdcm/nemesis.py#L1181

### SCT configuration:

https://github.com/scylladb/scylla-cluster-tests/blob/master/test-cases/longevity/longevity-10gb-3h.yaml

### Nemesis:

```
< t:2020-02-10 04:27:27 >>Started random_disrupt_method  network_reject_inter_node_communication
< t:2020-02-10 04:31:28 <<Finished random_disrupt_method network_reject_inter_node_communication
< t:2020-02-10 04:38:58 >>Started random_disrupt_method  nodetool_decommission
< t:2020-02-10 04:48:03 <<Finished random_disrupt_method nodetool_decommission
< t:2020-02-10 04:56:16 >>Started random_disrupt_method  snapshot_operations

2020-02-10T04:56:29+00:00  longevity-10gb-3h-master-db-node-d5361982-3 !INFO
| scylla: Segmentation fault on shard 0.
```
### Uploaded coredump: 
https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000.gz

### Database Error Log:

```
2020-02-10T04:56:29+00:00  longevity-10gb-3h-master-db-node-d5361982-3 !INFO
| scylla: Segmentation fault on shard 0.
| scylla: Backtrace:
| scylla: 0x0000000002d04102
| scylla: 0x0000000002ca5f1b
| scylla: 0x0000000002ca6215
| scylla: 0x0000000002ca6260
| scylla: 0x00007f9851934b1f
| scylla: 0x0000000002fa6b06
| scylla: 0x0000000002ca2110
| scylla: 0x0000000002ca231f
| scylla: 0x0000000002cddcc5
| scylla: 0x0000000002c3f82f
| scylla: 0x0000000002c4095e
| scylla: 0x0000000000c6e1a0
| scylla: /opt/scylladb/libreloc/libc.so.6+0x00000000000271a2
| scylla: 0x0000000000b9a02d
| kernel: scylla[3947]: segfault at 20 ip 0000000002fa6b07 sp 00007fff0b56ece0 error 4 in scylla[b9a000+2598000]
| kernel: Code: 8b 47 10 48 83 f8 03 0f 87 1e 01 00 00 0f 85 b6 01 00 00 48 8b 47 30 48 8b 77
 38 48 8d 7c 24 10 48 8b 50 08 c6 00 00 48 8b 06 <ff> 50 20 48 8b 44 24 18 4c 8d 65 18 48 83 f8 02 0f 87 83 00 00 00
```

### Decoded backtrace:

```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&
&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:743
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigsegv_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3465
 (inlined by) _FUN at ./build/release/seastar/./seastar/src/core/reactor.cc:3461
?? ??:0
seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, seastar::json::json_base_element const&)::{la
mbda()#2}::operator()() const at ./build/release/seastar/./seastar/src/json/json_elements.cc:116
 (inlined by) seastar::apply_helper<seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, seastar::
json::json_base_element const&)::{lambda()#2}, std::tuple<>&&, std::integer_sequence<unsigned long> >::apply({lambda()#2}&&, std::tuple<>) at ./build/release/seastar/./seast
ar/include/seastar/core/apply.hh:36
 (inlined by) auto seastar::apply<seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, seastar::js
on::json_base_element const&)::{lambda()#2}>(seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, 
seastar::json::json_base_element const&)::{lambda()#2}&&, std::tuple<>&&) at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) seastar::future<> seastar::futurize<seastar::future<> >::apply<seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>,
 std::allocator<char> > const&, seastar::json::json_base_element const&)::{lambda()#2}>(seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_tr
aits<char>, std::allocator<char> > const&, seastar::json::json_base_element const&)::{lambda()#2}&&, std::tuple<>&&) at ./build/release/seastar/./seastar/include/seastar/cor
e/future.hh:1645
 (inlined by) seastar::future<> seastar::future<>::then_impl<seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<
char> > const&, seastar::json::json_base_element const&)::{lambda()#2}, seastar::future<> >(seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::cha
r_traits<char>, std::allocator<char> > const&, seastar::json::json_base_element const&)::{lambda()#2}&&)::{lambda()#1}::operator()() const::{lambda(seastar::future_state<>&&
)#1}::operator()(seastar::future_state) at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1191
 (inlined by) seastar::continuation<seastar::future<> seastar::future<>::then_impl<seastar::json::json_stream_builder::add(std::__cxx11::basic_string<char, std::char_traits<
char>, std::allocator<char> > const&, seastar::json::json_base_element const&)::{lambda()#2}, seastar::future<> >(seastar::json::json_stream_builder::add(std::__cxx11::basic
_string<char, std::char_traits<char>, std::allocator<char> > const&, seastar::json::json_base_element const&)::{lambda()#2}&&)::{lambda()#1}::operator()() const::{lambda(sea
star::future_state<>&&)#1}>::run_and_dispose() at ./build/release/seastar/./seastar/include/seastar/core/future.hh:506
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2745
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
main at ./main.cc:485
?? ??:0
_start at ??:?
```

### Coredump

```
2020-02-10 04:56:29.000: (CoreDumpEvent Severity.CRITICAL): node=Node longevity-10gb-3h-master-db-node-d5361982-3 [34.245.180.237 | 10.0.0.152] (seed: False)
corefile_url=
https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000.gz
backtrace=           PID: 3947 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Mon 2020-02-10 04:56:29 UTC (2min 6s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: 8cd03e3733584c219a0dca67818d0c1e
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: longevity-10gb-3h-master-db-node-d5361982-3
      Coredump: /var/lib/systemd/coredump/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000
       Message: Process 3947 (scylla) of user 996 dumped core.
                
                Stack trace of thread 3947:
                #0  0x0000000002fa6b07 _ZZN7seastar4json19json_stream_builder3addERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS0_17json_base_elementEENKUlvE0_clEv (scylla)
                #1  0x0000000002ca2111 _ZN7seastar7reactor9run_tasksERNS0_10task_queueE (scylla)
                #2  0x0000000002ca2320 _ZN7seastar7reactor14run_some_tasksEv (scylla)
                #3  0x0000000002cddcc6 _ZN7seastar7reactor14run_some_tasksEv (scylla)
                #4  0x0000000002c3f830 _ZN7seastar12app_template14run_deprecatedEiPPcOSt8functionIFvvEE (scylla)
                #5  0x0000000002c4095f _ZN7seastar12app_template3runEiPPcOSt8functionIFNS_6futureIJiEEEvEE (scylla)
                #6  0x0000000000c6e1a1 main (scylla)
                #7  0x00007f9850eeb1a3 __libc_start_main (libc.so.6)
                #8  0x0000000000b9a02e _start (scylla)
                
                Stack trace of thread 3961:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3964:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3966:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3963:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3967:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3965:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3970:
                #0  0x00007f985193387c read (libpthread.so.0)
                #1  0x0000000002f341b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f344a2 operator() (scylla)
                #3  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f98519294e2 start_thread (libpthread.so.0)
                #5  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3952:
                #0  0x0000000002ca3565 _ZNKSt6vectorIPN7seastar17smp_message_queue9work_itemESaIS3_EE5emptyEv (scylla)
                #1  0x0000000002caac76 _ZN7seastar3smp11poll_queuesEv (scylla)
                #2  0x0000000002d08e66 _ZN7seastar7reactor10smp_pollfn4pollEv (scylla)
                #3  0x0000000002c9a56d _ZN7seastar7reactor9poll_onceEv (scylla)
                #4  0x0000000002cddcf9 _ZNKSt8functionIFbvEEclEv (scylla)
                #5  0x0000000002cec95d _ZZN7seastar3smp9configureEN5boost15program_options13variables_mapENS_14reactor_configEENKUlvE1_clEv (scylla)
                #6  0x0000000002c7432e _ZNKSt8functionIFvvEEclEv (scylla)
                #7  0x00007f98519294e2 start_thread (libpthread.so.0)
                #8  0x00007f9850fc56d3 __clone (libc.so.6)
                
                Stack trace of thread 3951:
                #0  0x000000000
```

download_instructions=
gsutil cp gs://upload.scylladb.com/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.8cd03e3733584c219a0dca67818d0c1e.3947.1581310589000000.gz

Restore Monitor Stack command:
$ hydra investigate show-monitor d5361982-d2a0-43cc-bc40-b0e725a7245e
Show all stored logs command:
$ hydra investigate show-logs d5361982-d2a0-43cc-bc40-b0e725a7245e

**Comment by amoskong** [comment]
The same two coredump occurred in https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-100gb-4h/115/ during snapshot_operation Nemesis.
Test-id: 62e7d74d-ebc0-4d5c-b11d-69120adf2588

```
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: Segmentation fault on shard 0.
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: Backtrace:
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x0000000002d04102
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x0000000002ca5f1b
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x0000000002ca6215
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x0000000002ca6260
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x00007f0385401b1f
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: 0x0000600000a4a8d7
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | kernel: scylla[4072]: segfault at 600000a4a8d8 ip 0000600000a4a8d8 sp 00007ffc8d804ae8    error 15
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | kernel: Code: 00 00 05 00 00 00 00 00 00 00 74 6f 74 61 6c 00 00 00 00 00 00 00 00 00 00  00 00 01 00 00 00 00 00 00 2e 69 60 a5 22 00 00 00 <58> 7c 5a 00 00 00 00 00 f0 a8 a4 00 00 60 00 00 04 00 00 00 00 00


2020-02-10 05:10:10.000: (DatabaseLogEvent Severity.CRITICAL): type=SEGMENTATION regex=segmentation line_number=11310 node=Node longevity-100gb-4h-master-db-node-62e7d74d-4
[34.241.202.4 | 10.0.39.152] (seed: False)
2020-02-10T05:10:10+00:00  longevity-100gb-4h-master-db-node-62e7d74d-4 !INFO    | scylla: Segmentation fault on shard 0.

void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&
&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:743
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigsegv_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3465
 (inlined by) _FUN at ./build/release/seastar/./seastar/src/core/reactor.cc:3461
?? ??:0
?? ??:0
```

```
2020-02-10 05:10:10.000: (CoreDumpEvent Severity.CRITICAL): node=Node longevity-100gb-4h-master-db-node-62e7d74d-4 [34.241.202.4 | 10.0.39.152] (seed: False)
corefile_url=
https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.4072.1581311410000000/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.
4072.1581311410000000.gz
backtrace=           PID: 4072 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Mon 2020-02-10 05:10:10 UTC (1min 53s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-t
o-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: cc86a24773d1409b9ac54a2d170dffaf
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: longevity-100gb-4h-master-db-node-62e7d74d-4
      Coredump: /var/lib/systemd/coredump/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.4072.1581311410000000
       Message: Process 4072 (scylla) of user 996 dumped core.
                
                Stack trace of thread 4072:
                #0  0x0000600000a4a8d8 n/a (n/a)
```

download_instructions=
gsutil cp gs://upload.scylladb.com/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.4072.1581311410000000/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.4072.1581311410000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.cc86a24773d1409b9ac54a2d170dffaf.4072.1581311410000000.gz

**Comment by slivne** [comment]
fixed by 8581617e783869ae8b57adf2ce007903ba0c9d2b
